engine/cleanupmanager/db.go

```diff
@@ -37,10 +43,22 @@ func NewDB(ctx context.Context, db *sql.DB) (*DB, error) {
 		db:   db,
 		lock: lock,
 
+		now:     p.P(`select now()`),
```

Comment:
Yes, we're using the database clock instead of the local machine, and more importantly using the timestamp of the current transaction.

It prevents issues if multiple engines have different times, as well as ensuring the entire cleanup happens from the perspective of a single moment in time.